contracts/contracts/ChatOracle.sol

```diff
@@ -228,6 +228,7 @@ contract ChatOracle {
         string memory errorMessage
     ) public onlyWhitelisted {
         require(!isPromptProcessed[promptId], "Prompt already processed");
+        isPromptProcessed[promptId] = true;
```

Comment:
Added this back still!
In case it succeeds without the IChatGpt call failure we don't need to do another tx.
Only call out `markPromptAsProcessed()` when tx gets reverted.

Reply:
Makes sense